modules/nginx-ingress-controller/main.tf

```diff
@@ -0,0 +1,109 @@
+
+
+data "aws_eks_cluster" "this" {
+  name = var.eks_cluster_name
+}
+
+data "aws_eks_cluster_auth" "this" {
+  name = var.eks_cluster_name
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = data.aws_eks_cluster.this.endpoint
+    cluster_ca_certificate = base64decode(data.aws_eks_cluster.this.certificate_authority.0.data)
+    token                  = data.aws_eks_cluster_auth.this.token
+    load_config_file       = false
+  }
+}
+
+provider "kubernetes" {
+  host                   = data.aws_eks_cluster.this.endpoint
+  cluster_ca_certificate = base64decode(data.aws_eks_cluster.this.certificate_authority.0.data)
+  token                  = data.aws_eks_cluster_auth.this.token
+  load_config_file       = false
+  version                = "~> 1.9"
+}
+
+
+
+resource "kubernetes_namespace" "nginx_ingress" {
+  count = var.ingress_controller_enable == true ? 1 : 0
+
+  metadata {
+    name = var.kubernetes_namespace
+  }
+
+}
+
+resource "null_resource" "update_helm_repo" {
```

Comment:
Not used - should be removed

Reply:
? it was added because i saw this issue https://github.com/hashicorp/terraform-provider-helm/issues/438

Reply:
Yeah, but you're not using `data.helm_repository` which is correct since it's deprecated. Instead you put the repo URL directly onto the `resource.helm_release.repository` in [main.tf#62](https://github.com/nuuday/terraform-aws-eks-addons/pull/18/files/dcdad4b1939b1a3f65b44604f7489b60d836b484#diff-35477f84d50b45a1983e6f88a2dc62acR62) which is _also_ correct :slightly_smiling_face:

Reply:
removed